pages/api/flashcards/[id].js

```diff
@@ -55,9 +63,14 @@ export default async function handler(request, response) {
 
   if (request.method === "DELETE") {
     try {
-      await Flashcard.findByIdAndDelete(id);
-      response.status(200).json({ message: "Flashcard deleted." });
-      return;
+      if (session) {
+        await Flashcard.findByIdAndDelete(id);
+        response.status(200).json({ message: "Flashcard deleted." });
+        return;
+      } else {
+        response.status(401).json({ status: "Not authorized" });
+        return;
+      }
```

Comment:
you could do the same here if you like:
```suggestion
      if (!session) {
        response.status(401).json({ status: "Not authorized" });
        return;
      }
      
      await Flashcard.findByIdAndDelete(id);
      response.status(200).json({ message: "Flashcard deleted." });
      return;
```

Reply:
I would suggest to use this pattern everywhere to keep the code structure predictable for readers and reduce the nestind / complexity at the same time.